[btbrandon] iP

btbrandon opened the pull request:
### BeeBot 🐝 🍯 🤖 

---
> "Plans are nothing; planning is everything." — Dwight D. Eisenhower

**BeeBot takes the load off your mind by managing your tasks efficiently. It’s:**

- Text-based ✅ 
- User-friendly ✅ 
- ~~Thunder~~ LIGHTNING fast ⚡ 

---

### To buzz your way to productivity:

1. Fork the BeeBot 🐝  Repository from [HERE](https://github.com/btbrandon/ip)
2. Clone the source code into your local IDE
3. Enter the command "./gradlew run" in the CLI

And yes, it's completely **FREE**!

---

### Features:
 
- [x] Task Management
- [x] Add save file saving and reading features
- [x] Standardised date formatting 
- [x] Add GUI

---

If you're a Java programmer, BeeBot also offers a great way to sweeten 🍯  your Java skills. Here's the main method: 
```
public static void main(String[] args) {
     Application.launch(Main.class, args);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

btbrandon committed:
Completed Level-0

btbrandon committed:
Completed Level-1

btbrandon committed:
Completed Level-2

btbrandon committed:
Completed Level-3

btbrandon committed:
Completed Level-4

btbrandon committed:
Completed A-TextUiTesting

btbrandon committed:
Completed Level-5

btbrandon committed:
Completed Level-6

btbrandon committed:
Completed Level-7

btbrandon committed:
Completed Level-8

btbrandon committed:
Completed A-MoreOOP

btbrandon committed:
Completed A-Packages

btbrandon committed:
Merge branch 'add-gradle-support'

btbrandon committed:
Complete A-Gradle

btbrandon committed:
Complete A-JUnit

btbrandon committed:
Complete A-Jar

btbrandon committed:
Complete A-JavaDoc

btbrandon committed:
Complete A-CodingStandard

btbrandon committed:
Complete Level-9

btbrandon committed:
Fix merge conflicts

btbrandon committed:
Fix merge conflicts

Tanziyanghenry commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+     * @return The concatenated string.
+     */
+    public static String concatenate(String[] parts, int start) {
+        StringBuilder result = new StringBuilder();
```
Is there a more specific name other than result?

Tanziyanghenry commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,151 @@
+     * @throws MissingDeadlineException If the delimiter is missing or the format is incorrect
+     */
+    public static String concatenateUntil(String[] parts, String delimiter) throws MissingDeadlineException {
+        StringBuilder result = new StringBuilder();
```
Is there a more specific name other than result?

Tanziyanghenry commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,72 @@
+ * </p>
+ */
+public class Task {
+    private boolean done = false;
```
will isDone be better than done?

btbrandon replied:
yes i have changed it! thanks for the comment

Tanziyanghenry commented on src/main/java/BeeBot.java:
```diff
@@ -0,0 +1,168 @@
+                    Parser.speak("Nice! I've marked this task as done:\n" + currTask.toString());
+                    break;
+                case "unmark":
+                    int taskNum2 = Integer.parseInt(parts[1]);
```
is there a better name other than taskNum2?

btbrandon replied:
yes i have changed it! thanks for the comment

Tanziyanghenry reviewed (commented):
Your code looks good to me!!

btbrandon committed:
Complete Level-10

ITZXITZX commented on src/main/java/BeeBot.java:
```diff
@@ -0,0 +1,151 @@
+        String cmd = parts[0];
+        try {
+            switch (cmd) {
+                case "list":
```
Should there be an indentation for the [case clause after the switch statement](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=The%20switch%20statement%20should%20have%20the%20following%20form%3A%20Note%20there%20is%20no%20indentation%20for%20case%20clauses.)?

ITZXITZX commented on src/main/java/tasklist/TaskList.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    public static void speakTaskRemoved(String name, int size) {
+        String msg = "Noted. I've removed this task:\n" + name;
+        msg += size == 1 ? "Now you have 1 task in the list.\n" : "Now you have " + size + " tasks in the list.\n";
```
Is this line length a bit [too long](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Line%20length%20should%20be%20no%20longer%20than%20120%20chars.)?

ITZXITZX commented on src/main/resources/view/MainWindow.fxml:
```diff
@@ -0,0 +1,44 @@
+    <children>
+        <TextField fx:id="userInput"
+                   layoutY="558.0"
+                   onAction="#handleUserInput"
```
I like how you formatted the code here 😄

ITZXITZX reviewed (commented):
Looks good overall, great work!

btbrandon committed:
Added assertions to BeeBot class for improved error checking

- Added assertions to verify that the file path provided in the constructor is neither null nor empty.
- Added assertions to ensure that the task numbers used for marking, unmarking and deleting are within the valid range of the list.

These assertions help to enforce assumptions about the validity of input, improving robustness and helping in debugging.

btbrandon committed:
Edited code to adhere to Code Quality Standards for improved code readability

- Changed boolean value to follow "is-" standard.
- Added clearer names for task numbers in Task class.

These changes help to increase clarity and improve code readability for users.

btbrandon committed:
Merge pull request #1 from btbrandon/branch-A-Assertions

Added assertions to BeeBot class for improved error checking

btbrandon committed:
Merge branch 'master' into branch-A-CodeQuality

btbrandon committed:
Merge pull request #2 from btbrandon/branch-A-CodeQuality

Edited code to adhere to Code Standards for improved code readability

btbrandon committed:
Add update function to BeeBot class to allow updating names

This increases flexibility for users by allowing them to directly edit the names of previously added tasks.

btbrandon committed:
Merge pull request #3 from btbrandon/branch-C-Update

Add update function to BeeBot class to allow updating names

btbrandon committed:
Complete A-UserGuide

btbrandon committed:
Update A-UserGuide

btbrandon committed:
Update BeeBot Coding Standards

btbrandon committed:
Update spaces between code

btbrandon committed:
Delete Ui class

btbrandon committed:
Update file path

btbrandon committed:
fix bugs: File loading

btbrandon committed:
Fix bug: Entry point
